core/src/main/java/org/web3j/protocol/websocket/WebSocketService.java

```diff
@@ -433,8 +433,16 @@ private long getReplyId(JsonNode replyJson) throws IOException {
         }
 
         if (!idField.isIntegralNumber()) {
-            throw new IOException(
-                    String.format("'id' expected to be long, but it is: '%s'", idField.asText()));
+            if (idField.isTextual()) {
+                try {
+                    return Long.parseLong(idField.asText());
+                } catch (Exception e) {
```

Comment:
Shouldn't it make more sense to catch only `NumberFormatException`s?

Reply:
I agree. I will update the Exception part. 

Also, I don't know how to `spotlessApply`, (I think it is about format of code, like spaces, structure, etc). I will search about it and see if i can do it. I have pulled the code and opened in intellij. Will see if reformat code does the job.